### ISSUE TITLE
SKUs not merged in getProducts and getSubscriptions

### ISSUE THREAD

**Issue by blaues0cke** [dooboolab/react-native-iap#230]
### Version of react-native-iap

1.3.6

### Platforms you faced the error (IOS or Android or both?)

iOS

### Expected behavior

Get a valid internal list merged with the results of `getProducts` and `getSubscriptions`

### Actual behavior

The internal `validProducts` only contains the last result set of `getProducts` or `getSubscriptions`. This will end up in app store calls resulting in "Error: Invalid product ID.".

### Tested environment (Emulator? Real Device?)

Tested on the iOS emulator and the real device.

### Steps to reproduce the behavior

I am running code like this with a "correct set" of SKUs in `productSkuList` and `subscriptionSkuList`:

```
const prepare = function* () {
    console.log('IAP: Preparing in app purchases');

    try {
        yield RNIap.prepare();

        const products      = yield RNIap.getProducts(productSkuList);
        const subscriptions = yield RNIap.getSubscriptions(subscriptionSkuList);

        yield RNIap.endConnection();
        yield put(PurchaseActions.prepareSuccess(products, subscriptions));
    } catch (error) {
        console.log('IAP: Error', error);

        yield RNIap.endConnection();
        yield put(PurchaseActions.prepareFailed());
    }
};
```
The products are loaded correctly but I am not able to run any `buyProduct`-call with SKUs from `productSkuList` since the internal array `validProducts` is overwritten by the different result from `getSubscriptions`. In my opinion this should be fixed or at least more detailed in the documentation since this will cost a lot of time to debug.

**Comment by JJMoon** [comment]
I am afraid that I don't understand your question.
But, I suggest you to use itemSkus which is defined by you in your source code. 
Please refer to the example code. https://github.com/dooboolab/react-native-iap/blob/master/RNIapExample/src/components/pages/First.js

**Comment by blaues0cke** [comment]
Sorry. I try to be more specific: 

1) I use `RNIap.getProducts(productSkuList)` to get a list of all purchasable products available for my app. This will make contain the objc-variable `validProducts` a valid list of products.
2) Then I call `RNIap.getSubscriptions(subscriptionSkuList)` to get a list of all subscriptions available for my app. This will **overwrite** the objc-variable `validProducts` with the new list of the subscriptions since there is no difference between `RNIap.getProducts()` and `RNIap.getSubscriptions()` on iOS.

But if you now try to call `RNIap.buyProduct()` with an SKU fetched earlier in `RNIap.getProducts()`, react-native-iap will fail with the error "Error: Invalid product ID.". since it thinks the product is invalid because it was overwritten by the second call of `RNIap.getSubscriptions()`. 

I lost about a day in this, just thought it would be nice to fix this/or warn the next person who is about to run into this. :-)

**Comment by JJMoon** [comment]
Oh. I got the point. I will fix it. 
Sorry for that.

**Comment by dooboolab** [comment]
@blaues0cke This will be fixed in `2.1.0`. Could you try this and come back again?

**Comment by blaues0cke** [comment]
@dooboolab For sure, will do that an come back to you here.
